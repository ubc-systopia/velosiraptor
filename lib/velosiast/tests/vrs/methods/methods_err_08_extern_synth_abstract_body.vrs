extern abstract fn foo() -> bool


extern synth fn foo() -> bool

extern fn foo() -> bool {
    true
}