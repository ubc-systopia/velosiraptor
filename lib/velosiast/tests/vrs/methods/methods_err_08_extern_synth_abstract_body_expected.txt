error: an extern method cannot be abstract
     --> tests/vrs/methods/methods_err_08_extern_synth_abstract_body.vrs:1:8
      |
    1 |  extern abstract fn foo() -> bool
      |         ^^^^^^^^ remove the abstract modifier
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: an extern method cannot be synth
     --> tests/vrs/methods/methods_err_08_extern_synth_abstract_body.vrs:4:8
      |
    4 |  extern synth fn foo() -> bool
      |         ^^^^^ remove the synth modifier
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: method defined as extern cannot have a body.
     --> tests/vrs/methods/methods_err_08_extern_synth_abstract_body.vrs:6:1
      |
    6 |  extern fn foo() -> bool {
      |  ^^^^^^ remove the `extern` modifier
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
